idseq_pipeline/commands/accessionid2seq_functions.py

```diff
@@ -234,29 +237,33 @@ def get_sequence_for_thread(error_flags, accession_info, accession_id, nt_loc_di
 
 def get_sequence_by_accession_id_ntf(accession_id, nt_loc_dict, ntf):
     ref_seq = ''
+    seq_name = ''
     entry = nt_loc_dict.get(accession_id)
     if entry:
-        range_start = entry[0]+entry[1]
-        seq_len = entry[2]
+        range_start = entry[0]
+        seq_len = entry[1] + entry[2]
         ntf.seek(range_start, 0)
-        ref_seq = ntf.read(seq_len).replace("\n", "")
-    return ref_seq
+        (seq_name, ref_seq) = ntf.read(seq_len).split("\n", 1)
+        ref_seq = ref_seq.replace("\n", "")
+        seq_name = seq_name.split(" ", 1)[1]
+    return (ref_seq, seq_name)
 
 def get_sequence_by_accession_id_s3(accession_id, nt_loc_dict, nt_bucket, nt_key):
     seq_len = 0
+    seq_name = ''
     entry = nt_loc_dict.get(accession_id)
     if entry:
-        range_start = entry[0]+entry[1]
-        seq_len = entry[2]
+        (range_start, name_length, seq_len) = entry
+
         accession_file = 'accession-%s' % accession_id
         NUM_RETRIES = 3
         for attempt in range(NUM_RETRIES):
             try:
                 pipe_file = 'pipe-{attempt}-accession-{accession_id}'.format(attempt=attempt, accession_id=accession_id)
                 os.mkfifo(pipe_file)
-                get_range = "aws s3api get-object --range bytes=%d-%d --bucket %s --key %s %s" % (range_start, range_start + seq_len - 1, nt_bucket, nt_key, pipe_file)
+                get_range = "aws s3api get-object --range bytes=%d-%d --bucket %s --key %s %s" % (range_start, range_start + name_length + seq_len - 1, nt_bucket, nt_key, pipe_file)
                 get_range_proc = subprocess.Popen(get_range, shell=True, stdout=DEVNULL)
-                subprocess.check_call("tr -d '\n' < {pipe_file} > {accession_file}".format(pipe_file=pipe_file, accession_file=accession_file), shell=True)
+                seq_name = subprocess.check_output("cat {pipe_file} |tee >(tail -n+2 |tr -d '\n' > {accession_file}) |head -1 ".format(pipe_file=pipe_file, accession_file=accession_file), executable='/bin/bash', shell=True).split(" ", 1)[1]
```

Comment:
This looks nice and clean.    I have some suspicions about `check_output` making it slow -- since that blocks the entire pipeline until the python process is ready to receive -- and was a problem in one of the earlier versions.  But with some luck the buffering in the pipe will be enough to ensure this works fine.

Could you run this on a big job, one that used to take 82 minutes, but now takes 8, and see it is still 8 after your change?

Reply:
which sample specifically?